frontend/packages/console-dynamic-plugin-sdk/src/extensions/navigation.ts

```diff
@@ -0,0 +1,69 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+
+export namespace ExtensionProperties {
+  export type NavItem = {
```

Comment:
We need a way to supply `data-*` properties to nav items.
Perhaps add `dataAttributes?: {[key: string]: string}`. Where each attribute gets the prefix `data-`
eg.
```
dataAttributes: {
  ['test-id']: 'my-id',
}
```

Results in: `<NavItem data-test-id="my-id"`

```
    /** Adds `data-` attributes to the DOM. Each key will receive the `data-` prefix.  */
    dataAttributes?: {[key: string]: string},
```